class/practice10/main.cpp

```diff
@@ -0,0 +1,201 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <random>
+#include <print>
+#include <ranges>
+#include <algorithm>
+class RandomGen{
+    public:
+        int randomInt(int min, int max) {
+            std::random_device rd;
+            std::mt19937 gen(rd());
+            std::uniform_int_distribution<> dis(min, max);
+            return dis(gen);
+        }
+
+        double randomDouble(double min, double max) {
+            std::random_device rd;
+            std::mt19937 gen(rd());
+            std::uniform_real_distribution<> dis(min, max);
+            return dis(gen);
+        }
+};
+
+class Team{
+    public:
+        Team(std::string name) : name(name), goalsScored(0), goalsAgainst(0), yellowCards(0), redCards(0), topHeight(0), points(0) {}
+
+        int goalDelta() const {
+            return goalsScored - goalsAgainst;
+        }
+        //Setters
+        void setGoalsScored(int goals){
+            goalsScored=goals;
+        }
+
+        void setGoalsAgainst(int goals){
+            goalsAgainst=goals;
+        }
+
+        void setYellowCards(int cards){
+            yellowCards=cards;
+        }
+
+        void setRedCards(int cards){
+            redCards=cards;
+        }
+
+        void setTopHeight(double height){
+            topHeight=height;
+        }
+
+        void setPoints(double newPoints){
+            points=newPoints;
+        }
+
+        //Getters
+        int getGoalsScored() const {
+            return goalsScored;
+        }
+
+        int getGoalsAgainst() const {
+            return goalsAgainst;
+        }
+
+        int getYellowCards() const {
+            return yellowCards;
+        }
+
+        int getRedCards() const {
+            return redCards;
+        }
+
+        double getTopHeight() const {
+            return topHeight;
+        }
+
+        double getPoints() const {
+            return points;
+        }
+
+        std::string getName() const {
+            return name;
+        }
+    private:
+        std::string name;
+        double points;
+        int goalsScored;
+        int goalsAgainst;
+        int yellowCards;
+        int redCards;
+        double topHeight;
+};
+
+
+class Match {
+public:
+    Team& first;
+    Team& second;
+    void playMatch() {
+        RandomGen rand;
+        int goalsFirst = rand.randomInt(0, 5);
+        int goalsSecond = rand.randomInt(0, 5);
+        int yellowCardsFirst = rand.randomInt(0, 3);
+        int yellowCardsSecond = rand.randomInt(0, 3);
+        int redCardsFirst = rand.randomInt(0, 2);
+        int redCardsSecond = rand.randomInt(0, 2);
+        double topHeightFirst = rand.randomDouble(0.0, 20.0);
+        double topHeightSecond = rand.randomDouble(0.0, 20.0);
+
+        first.setGoalsScored(first.getGoalsScored()+goalsFirst);
```

Comment:
i don't think we should use that getter there as we set the final goals count without increment scenario.
There should be some redesign I hope - maybe team should contain set of matches with exact results. When we accumulate goals and other info then we will loose some info related to each match and it will be hard to reuse it for other scenarios - for example viewing history of last few matches for the team and so on.
Therefore some redesign is needed. Tournament contains a list of teams and list of matches. Team can contain a list of references to matches for example. Or you can provide some other approach but the current one doesn't cover all the entities properly.